src/main/java/org/springframework/session/data/mongo/MongoOperationsSessionRepository.java

```diff
@@ -95,7 +95,19 @@ public MongoSession createSession() {
 
 	@Override
 	public void save(MongoSession session) {
-		this.mongoOperations.save(convertToDBObject(this.mongoSessionConverter, session), this.collectionName);
+
```

Comment:
The reported issue and this PR are for alignment with `WebSession` API, which is a reactive thing, so we likely shouldn't make changes to `MongoOperationsSessionRepository`.

We can revisit that depending on outcome of spring-projects/spring-session#1277 as ideally we'd make the behavior switch for all `SessionRepository` implementations in sync.